doc/source/parametertree/interactiveparameters.rst

```diff
@@ -0,0 +1,409 @@
+Interactive Parameters
+======================
+
+As indicated by the documentation for Parameters and Parameter Trees, a
+Parameter is commonly used to expose a value to the user without
+burdening developers with GUI representations. The ``interact`` method
+and friends extend such support to arbitrary Python functions. [1]_
+Before reading further, make sure to read existing Parameter
+documentation to become familiar with common extra options, creation
+techniques, and so on.
+
+Basic Use
+---------
+
+Consider a function ``a`` whose arguments should be tweakable by the
+user to immediately update some result. *Without* using ``interact``,
+you might do something like this:
+
+.. code:: python
+
+    from pyqtgraph.Qt import QtWidgets
+    import pyqtgraph as pg
+    from pyqtgraph.parametertree import Parameter, ParameterTree, parameterTypes as ptypes
+
+    def a(x=5, y=6):
+       QtWidgets.QMessageBox.information(None, 'Hello World', f'X is {x}, Y is {y}')
+
+    # -----------
+    # discussion is from here:
+    params = Parameter.create(name='"a" parameters', type='group', children=[
+       dict(name='x', type='int', value=5),
+       dict(name='y', type='int', value=6)
+    ])
+
+    def onChange(_param, _value):
+       a(**params)
+
+    for child in params.children():
+       child.sigValueChanged.connect(onChange)
+    # to here
+    # -----------
+
+    app = pg.mkQApp()
+    tree = ParameterTree()
+    tree.setParameters(params)
+    tree.show()
+    pg.exec()
+
+Notice in the ``-----`` comment block, lots of boilerplate and value
+duplication takes place. If an argument name changes, or the default
+value changes, the parameter definition must be independently updated as
+well. In (very common) cases like these, use ``interact`` instead (the
+code below is functionally equivalent to above):
+
+.. code:: python
+
+    from pyqtgraph.Qt import QtWidgets
+    import pyqtgraph as pg
+    from pyqtgraph.parametertree import Parameter, ParameterTree, interact
+
+    def a(x=5, y=6):
+       QtWidgets.QMessageBox.information(None, 'Hello World', f'X is {x}, Y is {y}')
+
+   # One line of code, no name/value duplication
```

Comment:
Nit-pick level 1-million: Add 1 space here, before the `#`, the rendered code-block has the wrong indentation currently.